rfc/rfc022-discovery-service.md

```diff
@@ -48,16 +48,21 @@ All presentations MUST conform to the Presentation Definition associated with th
 
 Presentations MUST be encoded in JWT format as string ([JWT]).
 
+The protocol identifies the server and client roles. A client can be a server and vice versa. The individual operations define what to do in certain cases.
+
 ### 3.1 Registration
 
-Clients can publish a Verifiable Presentation to the list by sending it in an HTTP POST to the service endpoint.
+Clients can publish a Verifiable Presentation to the server by sending it in an HTTP POST to the service endpoint.
 The HTTP request body MUST be a Verifiable Presentation (in JWT format). The content type of the request MUST be `application/json`.
 
 The server MUST validate the Verifiable Presentation as specified in section 4. If the validation fails, it MUST return a 400 Bad Request response.
-If the validation succeeds, the Verifiable Presentation MUST be added to the list and the server MUST return a 201 Created response.
+If the validation succeeds, the Verifiable Presentation MUST be added to the internal list and the server MUST return a 201 Created response.
+If the Verifiable Presentation is already on the list, the server MUST return a 304 Not Modified response.
```

Comment:
why?

How is equalness determined? VP ID is under control of the wallet (and thus might not be unique, although it should be), hash is complicated due to signature suite semantics or requires canonicalization.

Reply:
removed